kernel/sched.c

```diff
@@ -545,7 +545,17 @@ void z_thread_single_abort(struct k_thread *thread)
 	if (IS_ENABLED(CONFIG_SMP)) {
 		z_sched_abort(thread);
 	}
-
+	/* Past this point the thread is guaranteed to never to run again
+	 * on this CPU, and is not currently running (or will ever run again)
+	 * on another CPU.
+	 */
+#ifdef CONFIG_USERSPACE
+	/* Remove this thread from its memory domain, which takes
+	* it off the domain's thread list and possibly also arch-
+	* specific tasks.
+	*/
+	z_mem_domain_exit_thread(thread);
```

Comment:
AFAICS this is now called without spin lock, so I suppose we need to adapt the function documentation:

```
/* Called when thread aborts during teardown tasks. sched_spinlock is held */
void z_mem_domain_exit_thread(struct k_thread *thread)
```